src/middlewared/middlewared/plugins/chart_releases_linux/events.py

```diff
@@ -49,30 +53,74 @@ async def events(self, release_name):
     async def refresh_events_state(self, chart_release_name=None):
         filters = [['id', '=', chart_release_name]] if chart_release_name else []
         for chart_release in await self.middleware.call('chart.release.query', filters):
-            ChartReleaseService.CHART_RELEASES[chart_release['name']] = chart_release
+            async with LOCKS[chart_release['name']]:
+                ChartReleaseService.CHART_RELEASES[chart_release['name']] = {
+                    'data': chart_release,
+                    'poll': False,
+                }
+
+    @private
+    @functools.cache
+    def get_queue(self):
+        iterable = []
+        cur = 2
+        while cur <= self.MAX_SECONDS:
+            iterable.append(cur)
+            cur *= 2
+        return iterable
```

Comment:
You can make this a generator that yields 2, 4, ..., 1024, 1024, 1024... and then just iterate in `poll_chart_release_status`. Additionally, you can put that to `middlewared.utils.itertools` as other packages might need exponential back-off too.